entity_reference_override.module

```diff
@@ -0,0 +1,36 @@
+<?php
+
+/**
+ * @file
+ * The entity_reference_override hook implementations.
+ */
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function entity_reference_override_entity_presave(EntityInterface $entity) {
+  if (isset($entity->overwritten) && $entity->overwritten === TRUE) {
+    throw new \Exception('not allowed');
```

Comment:
Let's trigger a specific exception and make it a child of LogicException - if someone hits this there is code that needs fixing. Also the message should make it easy to know which entity has been saved - ie. we need the ID and type.